### ISSUE TITLE
learning (fiting) performance

### ISSUE THREAD

**Issue by amryounis** [keras-team/keras#9766]
Hi,

I am new to Keras with two trials centered around MNIST sequential model. The first one I used traditional approach while in the second I used functional one.
Performance-wise, the first was more than 10 fold faster than the functional
Even with that, the first gave accuracy about 97% while the second was about 50%

I am using CPU-based tensorflow backend. Any clues?
